app/uk/gov/hmrc/residencenilratebandcalculator/JsonBuilder.scala

```diff
@@ -42,32 +45,32 @@ class JsonBuilder @Inject()(rnrbConnector: RnrbConnector) {
     }
   }
 
-  def buildFromCacheMap(cacheMap: CacheMap): Future[Either[String, JsValue]] = {
+  def buildFromCacheMap(cacheMap: CacheMap): Future[Try[JsValue]] = {
     futureSchema.map {
-      case Right(schema) => {
+      case Success(schema) => {
         val incomingJson = Json.toJson(setKeys(cacheMap))
         val validationResult = validator.validate(schema, incomingJson).asEither
         validationResult match {
           case Left(error) => {
             val errorString = error.seq.flatMap(_._2).map(_.message).foldLeft(new StringBuilder())(_ append _).toString()
-            Left(errorString)
+            Failure(new JsonInvalidException(errorString))
           }
           case Right(json) => {
             if (dateOfDeathIsIneligible(cacheMap)) {
-              Left("Date of death is before eligibility date")
+              Failure(new JsonInvalidException("Date of death is before eligibility date"))
             } else {
-              Right(json)
+              Success(json)
             }
           }
         }
       }
-      case Left(error) => Left(error)
+      case Failure(exception) => Failure(exception)
     }
   }
 
-  def build(sessionConnector: SessionConnector)(implicit headerCarrier: HeaderCarrier): Future[Either[String, JsValue]] = {
+  def build(sessionConnector: SessionConnector)(implicit headerCarrier: HeaderCarrier) = {
     sessionConnector.fetch.flatMap {
-      case None => Future.successful(Left("No cache map returned"))
+      case None => Future.successful(Failure(new NoCacheMapException("No cache map.")))
```

Comment:
Can we have a more meaningful error?  The type, NoCacheMapException, suggests there's no cache map.
"Unable to retrieve cache map from SessionConnector" maybe?